guide/english/certifications/coding-interview-prep/data-structures/create-a-circular-queue/index.md

```diff
@@ -3,8 +3,79 @@ title: Create a Circular Queue
 ---
 ## Create a Circular Queue
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/certifications/coding-interview-prep/data-structures/create-a-circular-queue/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+### Method:
+- In this challenge we create a Circular Queue data structure.
+- First, we need to create an array of `size` with all elements set to `null`.
+- Then we create an equeue method, which moves the write pointer but doesnt exceed the read pointer.
+- The dequeue method on the other hand, moves the read pointer but doesnt exceed the write pointer.
+- Example:
+  - First, we create an array of length 5:
+  ```output
+  [null, null, null, null, null]
+  ^Read @ 0
+  ^Write @ 0
+  ```
+  - Then we enqueue `a`, `b`, and `c`:
+  ```
+  [a, b, c, null, null]
+  ^Read @ 0
+           ^Write @ 3
+  ```output
+  - Now we dequeue all the enqueued items:
+  ```output
+  [null, null, null, null, null]
+                    ^Read @ 3
+                    ^Write @ 3
+  ```output
+  - Finally, we enqueue `d`, `e` and `f`:
+  ```output
+  [f, null, null, d, e]
+                 ^Read @ 3
+     ^Write @ 1
+  ```
+### Solution:
+```js
+class CircularQueue {
+    constructor(size) {
+ 
+      this.queue = [];
+      this.read = 0;
+      this.write = 0;
+      this.max = size - 1;
+ 
+      while (size > 0) {
+         this.queue.push(null);
+         size--;
+      }
+ 
+    }
+ 
+    print() {
```

Comment:
Should this log the queue if named `print`?  If not, maybe use a different name?

Reply:
Thats provided provided by the challenge itself. It simply logs the whole queue array.